server.js

```diff
@@ -19,8 +39,33 @@ app.use(cors())
 app.use(bodyParser.json())
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find().sort({createdAt: 'desc'}).limit(20).exec()
+  res.json(thoughts)
+})
+
+app.post('/thoughts', async (req, res) => {
+  const {message} = req.body
+  const thoughts = new Thought({message})
+
+  try {
+    const saveThought = await thoughts.save()
+    res.status(201).json(saveThought)
+  } catch (err) {
+    res.status(400).json({message: 'could not save task to db', error:err.errors})
+  }
+})
+
+app.post('/thoughts/:id/like', async (req, res) => {
```

Comment:
consider changing this one to PUT instead of POST. 

"POST is always for creating a resource ( does not matter if it was duplicated ) PUT is for checking if resource is exists then update , else create new resource. PATCH is always for update a resource."

Reply:
Your way of thinking is right, but documentation said that it must be 'POST'.